solarforecastarbiter/io/reference_observations/eia.py

```diff
@@ -0,0 +1,137 @@
+import os
+import logging
+from functools import partial
+import pandas as pd
+
+from solarforecastarbiter.io.fetch import eia
+from solarforecastarbiter.io.reference_observations import (
+    common, default_forecasts)
+
+from requests.exceptions import HTTPError
+
+logger = logging.getLogger('reference_data')
+
+
+def initialize_site_observations(api, site):
+    """Creates an observation at the site.
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.api.APISession
+        API Session object, authenticated for the Reference user.
+    site : solarforecastarbiter.datamodel.Site
+        The site object for which to create the Observations.
+
+    Notes
+    -----
+    Currently only creates observations for net load [MW], but EIA contains
+    other variables that may be incorporated later (e.g. solar generation).
+
+    """
+
+    sfa_var = "net_load"
+    logger.info(f'Creating {sfa_var} at {site.name}')
+    try:
+        common.create_observation(api, site, sfa_var)
+    except HTTPError as e:
+        logger.error(f'Could not create Observation for "{sfa_var}" '
+                     f'at EIA site {site.name}')
+        logger.debug(f'Error: {e.response.text}')
+
+
+def initialize_site_forecasts(api, site):
+    """Creates a forecast at the site.
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.api.APISession
+        API Session object, authenticated for the Reference user.
+    site : solarforecastarbiter.datamodel.Site
+        The site object for which to create the Observations.
+
+    """
+    common.create_forecasts(api, site, ["net_load"],
+                            default_forecasts.TEMPLATE_FORECASTS)
+
+
+def fetch(api, site, start, end, *, eia_api_key):
+    """Retrieve observation data for a EIA site between start and end.
+
+    Parameters
+    ----------
+    api : solarforecastarbiter.io.APISession
+        Unused but conforms to common.update_site_observations call
+    site : solarforecastarbiter.datamodel.Site
+        Site object with the appropriate metadata.
+    start : datetime
+        The beginning of the period to request data for.
+    end : datetime
+        The end of the period to request data for.
+    eia_api_key : str
+        API key for api.eia.gov
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        All of the requested data as a single DataFrame.
+
+    Notes
+    -----
+    Currently only fetches observations for net load [MW], but EIA contains
+    other variables that may be incorporated later (e.g. solar generation).
+
+    """
+    try:
+        site_extra_params = common.decode_extra_parameters(site)
+    except ValueError:
+        return pd.DataFrame()
+    eia_site_id = site_extra_params['network_api_id']
+    series_id = f"EBA.{eia_site_id}.D.H"   # hourly net load (demand)
```

Comment:
can you add the solar gen variable name in a comment for future reference?

Reply:
Will do.